File: js/242_ValidAnagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if(s.length != t.length) return false;
    var len = s.length;
    var hash = new Array(26);
    for( var i in s ){
        if(!hash[s[i]]) hash[s[i]] = 1;
        else hash[s[i]]++;
    }
    for( var j in t ){
        if( !hash[t[j]] || hash[t[j]] < 0) return false;
        else hash[t[j]]--;
    }
    return true;
};

File: cpp/242_ValidAnagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()) return false;
        int len = s.length();
        int hash[26];
        memset(hash, 0, sizeof(int)*26);
        for( int i = 0; i < len; ++i ){
            hash[s[i] - 'a']++;
        }
        for( int i = 0; i < len; ++i ){
            if( --hash[t[i] - 'a'] < 0 ){
                return false;
            } 
        }
        return true;
    }
};